sunpy/map/maputils.py

```diff
@@ -1,7 +1,252 @@
+        Cartesian.
+
+    Returns
+    -------
+    contains_full_disk : `~bool`
+        Returns False if any of the edge pixels are less than one solar radius
+        away from the center of the Sun.
+    """
+    # Calculate all the edge pixels
+    edges = map_edges(smap)
+    edge_pixels = list(chain.from_iterable([edges["lhs"], edges["rhs"], edges["top"], edges["bottom"]]))
+    x = [p[0] for p in edge_pixels] * u.pix
+    y = [p[1] for p in edge_pixels] * u.pix
+
+    # Calculate the edge of the world
+    edge_of_world = smap.pixel_to_world(x, y)
+
+    # Calculate the distance of the edge of the world in solar radii
+    distance = u.R_sun * np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2) / smap.rsun_obs
+
+    # Test if all the edge pixels are more than one solar radius distant
+    # and that the whole map is not all off disk.
+    return np.all(distance > 1*u.R_sun) and ~is_all_off_disk(smap)
+
+
+def is_all_off_disk(smap):
+    """
+    Checks if the entire map is off the solar disk.
+
+    The check is performed by calculating the distance of every
+    pixel from the center of the Sun.  If they are all off-disk,
+    then the function returns True. Otherwise, the function
+    returns False.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    is_all_off_disk : `~bool`
+        Returns True if all map pixels are strictly more than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(~coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
+
+
+def is_all_on_disk(smap):
+    """
+    Checks if the entire map is on the solar disk.
+
+    The check is performed by calculating the distance of every pixel
+    from the center of the Sun.  If they are all on-disk, then the
+    function returns True. Otherwise, the function returns False.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
+
+    Returns
+    -------
+    is_all_on_disk : `~bool`
+        Returns True if all map pixels are strictly less than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
+
+
+def contains_limb(smap):
+    """
+    Checks if a map contains a portion of the solar limb.
+
+    The check is performed by calculating the distance of every pixel from
+    the center of the Sun.  If at least one pixel is on disk and at least one
+    pixel is off disk, the function returns True.  Otherwise, the function
+    returns False. Note that this function will also true if the entire solar
+    limb is within the field of view of the map.  Note also that in the case
+    of coronagraph images the limb itself need not be observed.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        The input map.
```

Comment:
```suggestion
        The input map.
```

I might be over zealous here.